src/controller/c/supervisor.c

```diff
@@ -2056,6 +2083,84 @@ const char *wb_supervisor_node_get_base_type_name(WbNodeRef node) {
   return wb_node_get_name(node->type);
 }
 
+WbFieldRef wb_supervisor_node_get_field_by_index(WbNodeRef node, int index) {
+  if (!robot_check_supervisor(__FUNCTION__))
+    return NULL;
+
+  if (!is_node_ref_valid(node)) {
+    if (!robot_is_quitting())
+      fprintf(stderr, "Error: %s() called with a NULL or invalid 'node' argument.\n", __FUNCTION__);
+    return NULL;
+  }
+  const int node_fields = wb_supervisor_node_get_number_of_fields(node);
```

Comment:
You also have to remove this line given that `node_fields` is no longer used and adjust the error message.